packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -58,35 +68,56 @@ const operatorToSqlComparator = {
   '==': '=' as const, // Exact match
   '!=': '!=' as const, // Does not match
   '=@': 'ilike' as const, // Contains sub string
+  '<': '<' as const, // Less than
+  '<=': '<=' as const, // Less than or equal
+  '>': '>' as const, // Greater than
+  '>=': '>=' as const, // Greater than or equal
 };
 type Operator = keyof typeof operatorToSqlComparator;
 
 const filterOperators = Object.keys(operatorToSqlComparator) as Operator[];
 
-const formatComparisonValue = (value: string | string[], operator: Operator) => {
+type Value = string | string[] | number | number[];
+
+const formatComparisonValue = (value: Value, operator: Operator) => {
   if (operator === '=@' && typeof value === 'string') {
     return `%${value}%`;
   }
 
   return value;
 };
 
-const convertValueToAdvancedCriteria = (operator: Operator, value: string | string[]) => {
+const formatValue = <T extends keyof EntityFilterQuery>(field: T, value: string) =>
+  isNumericField(field) ? parseFloat(value) : value;
+
+const convertValueToAdvancedCriteria = (
+  field: keyof EntityFilterQuery,
+  operator: Operator,
+  value: string,
+) => {
+  const formattedValue = value.includes(MULTIPLE_VALUES_DELIMITER)
+    ? (value
+        .split(MULTIPLE_VALUES_DELIMITER)
+        .map(val => formatValue(field, val)) as ExtractArrays<Value>)
+    : formatValue(field, value);
+
   // For equal operator, we do not need to specify comparison object.
   if (operator === '==') {
-    return value;
+    return formattedValue;
   }
 
   const comparator = operatorToSqlComparator[operator];
 
   return {
     comparator,
-    comparisonValue: formatComparisonValue(value, operator),
+    comparisonValue: formatComparisonValue(formattedValue, operator),
   };
 };
 
 const toFilterClause = (queryClause: string) => {
-  const operator = filterOperators.find(o => queryClause.includes(o));
+  const [operator] = filterOperators
+    .filter(o => queryClause.includes(o))
+    .sort(getSortByKey('length', { ascending: false }));
```

Comment:
Clever! Smells a little too clever, but it does work every time :-)

Reply:
Okay honestly I am confused 😂 what does this do??

Reply:
It expands the match to use the longest operator found, e.g. `>=` instead of just `>`. The first `filter` pass will match both of them, so then we need to decide which one to use.

Alternatively we could change the order of `filterOperators` so that the longest operators are listed first, then use the first matched operator. Or we could sort that array by desc length. But I think that this would be a bit less implicit than what we do here...

I will add a comment